econml/drlearner.py

```diff
@@ -0,0 +1,633 @@
+    multitask_model_cate : object of type(model_final)
+        An instance of the model_final object that was fitted after calling fit which corresponds whose
+        vector of outcomes correspond to the CATE model for each treatment, compared to baseline.
+        Available only when multitask_model_final=False.
+    featurizer : object of type(featurizer)
+        An instance of the fitted featurizer that was used to preprocess X in the final CATE model training.
+        Available only when featurizer is not None and X is not None.
+    cate_feature_names(input_feature_names=None) : list of feature names or None
+        A list of feature names that correspond to the input features in the final CATE model. If
+        input_feature_names is not None and featurizer is None, then the input_feature_names are returned.
+        If the featurizer is not None, then this attribute is available only when the featurizer has
+        a method: `get_feature_names(input_feature_names)`. Otherwise None is returned.
+    score_ : float
+        The MSE in the final doubly robust potential outcome regressions, i.e.
+
+        .. math::
+            \\frac{1}{n_t} \\sum_{t=1}^{n_t} \\frac{1}{n} \\sum_{i=1}^n (Y_{i, t}^{DR} - \\hat{\\theta}_t(X_i))^2
+
+        where n_t is the number of treatments (excluding control).
+
+        If `sample_weight` is not None at fit time, then a weighted average across samples is returned.
+
+
+    """
+
+    def __init__(self, model_propensity=LogisticRegressionCV(cv=3, solver='lbfgs', multi_class='auto'),
+                 model_regression=WeightedLassoCV(cv=3),
+                 model_final=StatsModelsLinearRegression(),
+                 multitask_model_final=False,
+                 featurizer=None,
+                 n_splits=2,
+                 random_state=None):
+        class ModelNuisance:
+            def __init__(self, model_propensity, model_regression):
+                self._model_propensity = model_propensity
+                self._model_regression = model_regression
+
+            def _combine(self, X, W):
+                return np.hstack([arr for arr in [X, W] if arr is not None])
+
+            def fit(self, Y, T, X=None, W=None, *, sample_weight=None):
+                assert np.ndim(Y) == 1, "Can only accept single dimensional outcomes Y! Use Y.ravel()."
+                if (X is None) and (W is None):
+                    raise AttributeError("At least one of X or W has to not be None!")
+                if np.any(np.all(T == 0, axis=0)) or (not np.any(np.all(T == 0, axis=1))):
+                    raise AttributeError("Provided crossfit folds contain training splits that " +
+                                         "don't contain all treatments")
+                XW = self._combine(X, W)
+                filtered_kwargs = _filter_none_kwargs(sample_weight=sample_weight)
+                self._model_propensity.fit(XW, np.matmul(T, np.arange(1, T.shape[1] + 1)).ravel().astype(int),

```

Comment:
Consider creating a utility method for untransforming one-hot treatments - similar logic seems to have been put into many different places.